src/vivarium_csu_alzheimers/data/consistent_rates.py

```diff
@@ -0,0 +1,486 @@
+            row["mean"] = np.mean(tt.iloc[0])
+            row["standard_error"] = (
+                np.std(tt.iloc[0]) + 1e-8  # small epsilon to avoid zero standard errors
+            )
+
+            results.append(row)
+
+    return pd.DataFrame(results)
+
+
+def write_or_replace(art: Artifact, key: str, data: pd.DataFrame):
+    # Use artifact containment check for compatibility
+    if key in art:
+        art.replace(key, data)
+    else:
+        art.write(key, data)
+
+
+def generate_consistent_susceptible_to_bbbm_transition_count(art):
+    """use the cause.alzheimers_consistent.population_incidence_any and the time-varying population
+    to create a SUSCEPTIBLE_TO_BBBM_TRANSITION_COUNT value
+    """
+    prevalence = art.load(data_keys.ALZHEIMERS_CONSISTENT.PREVALENCE_ANY)
+    write_or_replace(art, data_keys.POPULATION.SCALING_FACTOR, prevalence)
+
+    pop = art.load("population.structure")
+
+    transition_rate = art.load("cause.alzheimers_consistent.population_incidence_any")
+
+    df = pd.merge(
+        pop.reset_index().drop(["location"], axis=1),
+        transition_rate.reset_index().drop(["year_start", "year_end"], axis=1),
+        on=["sex", "age_start", "age_end"],
+        suffixes=("", "_rate"),
+    ).set_index(["sex", "age_start", "age_end", "year_start", "year_end"])
+
+    df = pd.merge(
+        df.reset_index(),
+        prevalence.reset_index().drop(["year_start", "year_end"], axis=1),
+        on=["sex", "age_start", "age_end"],
+        suffixes=("", "_prev"),
+    ).set_index(["sex", "age_start", "age_end", "year_start", "year_end"])
+
+    for i in range(500):
+        df[f"draw_{i}"] *= df[f"draw_{i}_rate"] * (1 - df[f"draw_{i}_prev"])
+        del df[f"draw_{i}_rate"]
+        del df[f"draw_{i}_prev"]
+
+    write_or_replace(
+        art, "cause.alzheimers_consistent.susceptible_to_bbbm_transition_count", df
+    )
+
+
+def generate_consistent_dementia_conditional_prevalence(art):
+    bbbm_conditional_prevalence = art.load(
+        "cause.alzheimers_consistent.bbbm_conditional_prevalence"
+    )
+    mci_conditional_prevalence = art.load(
+        "cause.alzheimers_consistent.mci_conditional_prevalence"
+    )
+
+    dementia_conditional_prevalence = np.clip(
+        1 - (bbbm_conditional_prevalence + mci_conditional_prevalence), 0, 1
+    )
+
+    write_or_replace(
+        art,
+        "cause.alzheimers_consistent.dementia_conditional_prevalence",
+        dementia_conditional_prevalence,
+    )
+
+
+def generate_consistent_csmr(art):
+    emr = art.load("cause.alzheimers_consistent.excess_mortality_rate")
+    p = art.load("cause.alzheimers_consistent.prevalence_any")
+    dementia_conditional_prevalence = art.load(
+        "cause.alzheimers_consistent.dementia_conditional_prevalence"
+    )
+
+    csmr = emr * p * dementia_conditional_prevalence
+    write_or_replace(
+        art,
+        "cause.alzheimers_disease_and_other_dementias.cause_specific_mortality_rate",
+        csmr,
+    )
```

Comment:
I think I will have to take your word that the methodology and math in this module works out.